// @swimlane/ngx-datatable/esm2015/lib/components/footer/footer.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Input, Directive, TemplateRef, ContentChild } from '@angular/core';
import { DataTableFooterTemplateDirective } from './footer-template.directive';
import * as ɵngcc0 from '@angular/core';
export class DatatableFooterDirective {
    /**
     * @return {?}
     */
    get template() {
        return this._templateInput || this._templateQuery;
    }
}
DatatableFooterDirective.ɵfac = function DatatableFooterDirective_Factory(t) { return new (t || DatatableFooterDirective)(); };
DatatableFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DatatableFooterDirective, selectors: [["ngx-datatable-footer"]], contentQueries: function DatatableFooterDirective_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, DataTableFooterTemplateDirective, 1, TemplateRef);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx._templateQuery = _t.first);
    } }, inputs: { footerHeight: "footerHeight", totalMessage: "totalMessage", selectedMessage: "selectedMessage", pagerLeftArrowIcon: "pagerLeftArrowIcon", pagerRightArrowIcon: "pagerRightArrowIcon", pagerPreviousIcon: "pagerPreviousIcon", pagerNextIcon: "pagerNextIcon", _templateInput: ["template", "_templateInput"] } });
DatatableFooterDirective.propDecorators = {
    footerHeight: [{ type: Input }],
    totalMessage: [{ type: Input }],
    selectedMessage: [{ type: Input }],
    pagerLeftArrowIcon: [{ type: Input }],
    pagerRightArrowIcon: [{ type: Input }],
    pagerPreviousIcon: [{ type: Input }],
    pagerNextIcon: [{ type: Input }],
    _templateInput: [{ type: Input, args: ['template',] }],
    _templateQuery: [{ type: ContentChild, args: [DataTableFooterTemplateDirective, { read: TemplateRef, static: false },] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DatatableFooterDirective, [{
        type: Directive,
        args: [{ selector: 'ngx-datatable-footer' }]
    }], null, { footerHeight: [{
            type: Input
        }], totalMessage: [{
            type: Input
        }], selectedMessage: [{
            type: Input
        }], pagerLeftArrowIcon: [{
            type: Input
        }], pagerRightArrowIcon: [{
            type: Input
        }], pagerPreviousIcon: [{
            type: Input
        }], pagerNextIcon: [{
            type: Input
        }], _templateInput: [{
            type: Input,
            args: ['template']
        }], _templateQuery: [{
            type: ContentChild,
            args: [DataTableFooterTemplateDirective, { read: TemplateRef, static: false }]
        }] }); })();
if (false) {
    /** @type {?} */
    DatatableFooterDirective.prototype.footerHeight;
    /** @type {?} */
    DatatableFooterDirective.prototype.totalMessage;
    /** @type {?} */
    DatatableFooterDirective.prototype.selectedMessage;
    /** @type {?} */
    DatatableFooterDirective.prototype.pagerLeftArrowIcon;
    /** @type {?} */
    DatatableFooterDirective.prototype.pagerRightArrowIcon;
    /** @type {?} */
    DatatableFooterDirective.prototype.pagerPreviousIcon;
    /** @type {?} */
    DatatableFooterDirective.prototype.pagerNextIcon;
    /** @type {?} */
    DatatableFooterDirective.prototype._templateInput;
    /** @type {?} */
    DatatableFooterDirective.prototype._templateQuery;
}
