// angular2_photoswipe/bundles/angular2_photoswipe.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/common'), require('photoswipe'), require('photoswipe/dist/photoswipe-ui-default'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('angular2_photoswipe', ['@angular/core','@angular/common','exports', '@angular/common', 'photoswipe', 'photoswipe/dist/photoswipe-ui-default', '@angular/core'], factory) :
    (factory(global.ng.core,global.ng.common,(global.angular2_photoswipe = {}),global.ng.common,global.PhotoSwipe,global.PhotoSwipeUI_Default,global.ng.core));
}(this, (function (ɵngcc0,ɵngcc1,exports,common,PhotoSwipe,PhotoSwipeUI_Default,i0) { 
var _c0 = ["ngpLightbox"];
var _c1 = function (a0) { return { "card-img-top": a0 }; };
var _c2 = ["ngpGallery"];
var _c3 = [[["ngp-gallery-item"]]];
var _c4 = function (a0) { return { "card-columns": a0 }; };
var _c5 = ["ngp-gallery-item"];
'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var Angular2PhotoswipeService = /** @class */ (function () {
        function Angular2PhotoswipeService() {
        }
        /** @nocollapse */
        Angular2PhotoswipeService.ctorParameters = function () { return []; };
        /** @nocollapse */ Angular2PhotoswipeService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function Angular2PhotoswipeService_Factory() { return new Angular2PhotoswipeService(); }, token: Angular2PhotoswipeService, providedIn: "root" });
Angular2PhotoswipeService.ɵfac = function Angular2PhotoswipeService_Factory(t) { return new (t || Angular2PhotoswipeService)(); };
Angular2PhotoswipeService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: Angular2PhotoswipeService, factory: function (t) { return Angular2PhotoswipeService.ɵfac(t); }, providedIn: 'root' });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Angular2PhotoswipeService, [{
        type: i0.Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
        return Angular2PhotoswipeService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var Angular2PhotoswipeComponent = /** @class */ (function () {
        function Angular2PhotoswipeComponent() {
        }
        /**
         * @return {?}
         */
        Angular2PhotoswipeComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        /** @nocollapse */
        Angular2PhotoswipeComponent.ctorParameters = function () { return []; };
Angular2PhotoswipeComponent.ɵfac = function Angular2PhotoswipeComponent_Factory(t) { return new (t || Angular2PhotoswipeComponent)(); };
Angular2PhotoswipeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: Angular2PhotoswipeComponent, selectors: [["ngp-angular2-photoswipe"]], decls: 2, vars: 0, template: function Angular2PhotoswipeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "p");
        ɵngcc0.ɵɵtext(1, " angular2-photoswipe works! ");
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Angular2PhotoswipeComponent, [{
        type: i0.Component,
        args: [{
                selector: 'ngp-angular2-photoswipe',
                template: "\n    <p>\n      angular2-photoswipe works!\n    </p>\n  "
            }]
    }], function () { return []; }, null); })();
        return Angular2PhotoswipeComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgpService = /** @class */ (function () {
        function NgpService() {
        }
        /** @nocollapse */ NgpService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NgpService_Factory() { return new NgpService(); }, token: NgpService, providedIn: "root" });
NgpService.ɵfac = function NgpService_Factory(t) { return new (t || NgpService)(); };
NgpService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: NgpService, factory: function (t) { return NgpService.ɵfac(t); }, providedIn: 'root' });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgpService, [{
        type: i0.Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
        return NgpService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var LightboxComponent = /** @class */ (function () {
        function LightboxComponent(ngp) {
            this.ngp = ngp;
        }
        /**
         * @return {?}
         */
        LightboxComponent.prototype.ngAfterContentInit = /**
         * @return {?}
         */
            function () {
                this.ngp.LightboxElement = this.el;
            };
        /** @nocollapse */
        LightboxComponent.ctorParameters = function () {
            return [
                { type: NgpService }
            ];
        };
        LightboxComponent.propDecorators = {
            el: [{ type: i0.ViewChild, args: ['ngpLightbox', { static: true },] }]
        };
LightboxComponent.ɵfac = function LightboxComponent_Factory(t) { return new (t || LightboxComponent)(ɵngcc0.ɵɵdirectiveInject(NgpService)); };
LightboxComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: LightboxComponent, selectors: [["ngp-lightbox"]], viewQuery: function LightboxComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 3);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.el = _t.first);
    } }, decls: 25, vars: 0, consts: [["tabindex", "-1", "role", "dialog", "aria-hidden", "true", 1, "pswp"], ["ngpLightbox", ""], [1, "pswp__bg"], [1, "pswp__scroll-wrap"], [1, "pswp__container"], [1, "pswp__item"], [1, "pswp__ui", "pswp__ui--hidden"], [1, "pswp__top-bar"], [1, "pswp__counter"], ["title", "Close (Esc)", 1, "pswp__button", "pswp__button--close"], ["title", "Share", 1, "pswp__button", "pswp__button--share"], ["title", "Toggle fullscreen", 1, "pswp__button", "pswp__button--fs"], ["title", "Zoom in/out", 1, "pswp__button", "pswp__button--zoom"], [1, "pswp__preloader"], [1, "pswp__preloader__icn"], [1, "pswp__preloader__cut"], [1, "pswp__preloader__donut"], [1, "pswp__share-modal", "pswp__share-modal--hidden", "pswp__single-tap"], [1, "pswp__share-tooltip"], ["title", "Previous (arrow left)", 1, "pswp__button", "pswp__button--arrow--left"], ["title", "Next (arrow right)", 1, "pswp__button", "pswp__button--arrow--right"], [1, "pswp__caption"], [1, "pswp__caption__center"]], template: function LightboxComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0, 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵelementStart(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelement(6, "div", 5);
        ɵngcc0.ɵɵelement(7, "div", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "div", 6);
        ɵngcc0.ɵɵelementStart(9, "div", 7);
        ɵngcc0.ɵɵelement(10, "div", 8);
        ɵngcc0.ɵɵelement(11, "button", 9);
        ɵngcc0.ɵɵelement(12, "button", 10);
        ɵngcc0.ɵɵelement(13, "button", 11);
        ɵngcc0.ɵɵelement(14, "button", 12);
        ɵngcc0.ɵɵelementStart(15, "div", 13);
        ɵngcc0.ɵɵelementStart(16, "div", 14);
        ɵngcc0.ɵɵelementStart(17, "div", 15);
        ɵngcc0.ɵɵelement(18, "div", 16);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 17);
        ɵngcc0.ɵɵelement(20, "div", 18);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(21, "button", 19);
        ɵngcc0.ɵɵelement(22, "button", 20);
        ɵngcc0.ɵɵelementStart(23, "div", 21);
        ɵngcc0.ɵɵelement(24, "div", 22);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } }, styles: [".pswp[_ngcontent-%COMP%]{display:none;position:absolute;width:100%;height:100%;left:0;top:0;overflow:hidden;touch-action:none;z-index:1500;-webkit-text-size-adjust:100%;-webkit-backface-visibility:hidden;outline:0}.pswp[_ngcontent-%COMP%]   *[_ngcontent-%COMP%]{box-sizing:border-box}.pswp[_ngcontent-%COMP%]   img[_ngcontent-%COMP%]{max-width:none}.pswp--animate_opacity[_ngcontent-%COMP%]{opacity:.001;will-change:opacity;transition:opacity 333ms cubic-bezier(.4,0,.22,1)}.pswp--open[_ngcontent-%COMP%]{display:block}.pswp--zoom-allowed[_ngcontent-%COMP%]   .pswp__img[_ngcontent-%COMP%]{cursor:-webkit-zoom-in;cursor:-moz-zoom-in;cursor:zoom-in}.pswp--zoomed-in[_ngcontent-%COMP%]   .pswp__img[_ngcontent-%COMP%]{cursor:-webkit-grab;cursor:-moz-grab;cursor:grab}.pswp--dragging[_ngcontent-%COMP%]   .pswp__img[_ngcontent-%COMP%]{cursor:-webkit-grabbing;cursor:-moz-grabbing;cursor:grabbing}.pswp__bg[_ngcontent-%COMP%]{position:absolute;left:0;top:0;width:100%;height:100%;background:#000;opacity:0;-webkit-transform:translateZ(0);transform:translateZ(0);-webkit-backface-visibility:hidden;will-change:opacity}.pswp__scroll-wrap[_ngcontent-%COMP%]{position:absolute;left:0;top:0;width:100%;height:100%;overflow:hidden}.pswp__container[_ngcontent-%COMP%], .pswp__zoom-wrap[_ngcontent-%COMP%]{touch-action:none;position:absolute;left:0;right:0;top:0;bottom:0;-webkit-backface-visibility:hidden}.pswp__container[_ngcontent-%COMP%], .pswp__img[_ngcontent-%COMP%]{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none}.pswp__zoom-wrap[_ngcontent-%COMP%]{position:absolute;width:100%;-webkit-transform-origin:left top;transform-origin:left top;transition:transform 333ms cubic-bezier(.4,0,.22,1),-webkit-transform 333ms cubic-bezier(.4,0,.22,1)}.pswp__bg[_ngcontent-%COMP%]{will-change:opacity;transition:opacity 333ms cubic-bezier(.4,0,.22,1)}.pswp--animated-in[_ngcontent-%COMP%]   .pswp__bg[_ngcontent-%COMP%], .pswp--animated-in[_ngcontent-%COMP%]   .pswp__zoom-wrap[_ngcontent-%COMP%]{transition:none}.pswp__item[_ngcontent-%COMP%]{position:absolute;left:0;right:0;top:0;bottom:0;overflow:hidden}.pswp__img[_ngcontent-%COMP%]{position:absolute;width:auto;height:auto;top:0;left:0}.pswp__img--placeholder[_ngcontent-%COMP%]{-webkit-backface-visibility:hidden}.pswp__img--placeholder--blank[_ngcontent-%COMP%]{background:#222}.pswp--ie[_ngcontent-%COMP%]   .pswp__img[_ngcontent-%COMP%]{width:100%!important;height:auto!important;left:0;top:0}.pswp__error-msg[_ngcontent-%COMP%]{position:absolute;left:0;top:50%;width:100%;text-align:center;font-size:14px;line-height:16px;margin-top:-8px;color:#ccc}.pswp__error-msg[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]{color:#ccc;text-decoration:underline}.pswp__button[_ngcontent-%COMP%]{position:relative;cursor:pointer;overflow:visible;-webkit-appearance:none;display:block;border:0;padding:0;margin:0;float:right;opacity:.75;transition:opacity .2s;box-shadow:none}.pswp__button[_ngcontent-%COMP%]:focus, .pswp__button[_ngcontent-%COMP%]:hover{opacity:1}.pswp__button[_ngcontent-%COMP%]:active{outline:0;opacity:.9}.pswp__button[_ngcontent-%COMP%]::-moz-focus-inner{padding:0;border:0}.pswp__ui--over-close[_ngcontent-%COMP%]   .pswp__button--close[_ngcontent-%COMP%]{opacity:1}.pswp__button[_ngcontent-%COMP%], .pswp__button--arrow--left[_ngcontent-%COMP%]:before, .pswp__button--arrow--right[_ngcontent-%COMP%]:before{background:url(/assets/media/default-skin.png) 0 0/264px 88px no-repeat;width:44px;height:44px}@media (-webkit-min-device-pixel-ratio:1.1),(-webkit-min-device-pixel-ratio:1.09375),(min-resolution:105dpi),(min-resolution:1.1dppx){.pswp--svg[_ngcontent-%COMP%]   .pswp__button[_ngcontent-%COMP%], .pswp--svg[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%]:before, .pswp--svg[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%]:before{background-image:url(/assets/media/default-skin.svg)}.pswp--svg[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp--svg[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%]{background:0 0}}.pswp__button--close[_ngcontent-%COMP%]{background-position:0 -44px}.pswp__button--share[_ngcontent-%COMP%]{background-position:-44px -44px}.pswp__button--fs[_ngcontent-%COMP%]{display:none}.pswp--supports-fs[_ngcontent-%COMP%]   .pswp__button--fs[_ngcontent-%COMP%]{display:block}.pswp--fs[_ngcontent-%COMP%]   .pswp__button--fs[_ngcontent-%COMP%]{background-position:-44px 0}.pswp__button--zoom[_ngcontent-%COMP%]{display:none;background-position:-88px 0}.pswp--zoom-allowed[_ngcontent-%COMP%]   .pswp__button--zoom[_ngcontent-%COMP%]{display:block}.pswp--zoomed-in[_ngcontent-%COMP%]   .pswp__button--zoom[_ngcontent-%COMP%]{background-position:-132px 0}.pswp--touch[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp--touch[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%]{visibility:hidden}.pswp__button--arrow--left[_ngcontent-%COMP%], .pswp__button--arrow--right[_ngcontent-%COMP%]{background:0 0;top:50%;margin-top:-50px;width:70px;height:100px;position:absolute}.pswp__button--arrow--left[_ngcontent-%COMP%]{left:0}.pswp__button--arrow--right[_ngcontent-%COMP%]{right:0}.pswp__button--arrow--left[_ngcontent-%COMP%]:before, .pswp__button--arrow--right[_ngcontent-%COMP%]:before{content:'';top:35px;background-color:rgba(0,0,0,.3);height:30px;width:32px;position:absolute}.pswp__button--arrow--left[_ngcontent-%COMP%]:before{left:6px;background-position:-138px -44px}.pswp__button--arrow--right[_ngcontent-%COMP%]:before{right:6px;background-position:-94px -44px}.pswp__counter[_ngcontent-%COMP%], .pswp__share-modal[_ngcontent-%COMP%]{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.pswp__share-modal[_ngcontent-%COMP%]{display:block;background:rgba(0,0,0,.5);width:100%;height:100%;top:0;left:0;padding:10px;position:absolute;z-index:1600;opacity:0;transition:opacity .25s ease-out;-webkit-backface-visibility:hidden;will-change:opacity}.pswp__share-modal--hidden[_ngcontent-%COMP%]{display:none}.pswp__share-tooltip[_ngcontent-%COMP%]{z-index:1620;position:absolute;background:#fff;top:56px;border-radius:2px;display:block;width:auto;right:44px;box-shadow:0 2px 5px rgba(0,0,0,.25);-webkit-transform:translateY(6px);transform:translateY(6px);transition:transform .25s;transition:transform .25s,-webkit-transform .25s;-webkit-backface-visibility:hidden;will-change:transform}.pswp__share-tooltip[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]{display:block;padding:8px 12px;color:#000;text-decoration:none;font-size:14px;line-height:18px}.pswp__share-tooltip[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:hover{text-decoration:none;color:#000}.pswp__share-tooltip[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:first-child{border-radius:2px 2px 0 0}.pswp__share-tooltip[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]:last-child{border-radius:0 0 2px 2px}.pswp__share-modal--fade-in[_ngcontent-%COMP%]{opacity:1}.pswp__share-modal--fade-in[_ngcontent-%COMP%]   .pswp__share-tooltip[_ngcontent-%COMP%]{-webkit-transform:translateY(0);transform:translateY(0)}.pswp--touch[_ngcontent-%COMP%]   .pswp__share-tooltip[_ngcontent-%COMP%]   a[_ngcontent-%COMP%]{padding:16px 12px}a.pswp__share--facebook[_ngcontent-%COMP%]:before{content:'';display:block;width:0;height:0;position:absolute;top:-12px;right:15px;border:6px solid transparent;border-bottom-color:#fff;-webkit-pointer-events:none;-moz-pointer-events:none;pointer-events:none}a.pswp__share--facebook[_ngcontent-%COMP%]:hover{background:#3e5c9a;color:#fff}a.pswp__share--facebook[_ngcontent-%COMP%]:hover:before{border-bottom-color:#3e5c9a}a.pswp__share--twitter[_ngcontent-%COMP%]:hover{background:#55acee;color:#fff}a.pswp__share--pinterest[_ngcontent-%COMP%]:hover{background:#ccc;color:#ce272d}a.pswp__share--download[_ngcontent-%COMP%]:hover{background:#ddd}.pswp__counter[_ngcontent-%COMP%]{position:absolute;left:0;top:0;height:44px;font-size:13px;line-height:44px;color:#fff;opacity:.75;padding:0 10px}.pswp__caption[_ngcontent-%COMP%]{position:absolute;left:0;bottom:0;width:100%;min-height:44px}.pswp__caption[_ngcontent-%COMP%]   small[_ngcontent-%COMP%]{font-size:11px;color:#bbb}.pswp__caption__center[_ngcontent-%COMP%]{text-align:left;max-width:420px;margin:0 auto;font-size:13px;padding:10px;line-height:20px;color:#ccc}.pswp__caption--empty[_ngcontent-%COMP%]{display:none}.pswp__caption--fake[_ngcontent-%COMP%]{visibility:hidden}.pswp__preloader[_ngcontent-%COMP%]{width:44px;height:44px;position:absolute;top:0;left:50%;margin-left:-22px;opacity:0;transition:opacity .25s ease-out;will-change:opacity;direction:ltr}.pswp__preloader__icn[_ngcontent-%COMP%]{width:20px;height:20px;margin:12px}.pswp__preloader--active[_ngcontent-%COMP%]{opacity:1}.pswp__preloader--active[_ngcontent-%COMP%]   .pswp__preloader__icn[_ngcontent-%COMP%]{background:url(/assets/media/preloader.gif) no-repeat}.pswp--css_animation[_ngcontent-%COMP%]   .pswp__preloader--active[_ngcontent-%COMP%]{opacity:1}.pswp--css_animation[_ngcontent-%COMP%]   .pswp__preloader--active[_ngcontent-%COMP%]   .pswp__preloader__icn[_ngcontent-%COMP%]{-webkit-animation:.5s linear infinite clockwise;animation:.5s linear infinite clockwise}.pswp--css_animation[_ngcontent-%COMP%]   .pswp__preloader--active[_ngcontent-%COMP%]   .pswp__preloader__donut[_ngcontent-%COMP%]{-webkit-animation:1s cubic-bezier(.4,0,.22,1) infinite donut-rotate;animation:1s cubic-bezier(.4,0,.22,1) infinite donut-rotate}.pswp--css_animation[_ngcontent-%COMP%]   .pswp__preloader__icn[_ngcontent-%COMP%]{background:0 0;opacity:.75;width:14px;height:14px;position:absolute;left:15px;top:15px;margin:0}.pswp--css_animation[_ngcontent-%COMP%]   .pswp__preloader__cut[_ngcontent-%COMP%]{position:relative;width:7px;height:14px;overflow:hidden}.pswp--css_animation[_ngcontent-%COMP%]   .pswp__preloader__donut[_ngcontent-%COMP%]{box-sizing:border-box;width:14px;height:14px;border:2px solid #fff;border-radius:50%;border-left-color:transparent;border-bottom-color:transparent;position:absolute;top:0;left:0;background:0 0;margin:0}@media screen and (max-width:1024px){.pswp__preloader[_ngcontent-%COMP%]{position:relative;left:auto;top:auto;margin:0;float:right}}@-webkit-keyframes clockwise{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes clockwise{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes donut-rotate{0%,100%{-webkit-transform:rotate(0);transform:rotate(0)}50%{-webkit-transform:rotate(-140deg);transform:rotate(-140deg)}}@keyframes donut-rotate{0%,100%{-webkit-transform:rotate(0);transform:rotate(0)}50%{-webkit-transform:rotate(-140deg);transform:rotate(-140deg)}}.pswp__ui[_ngcontent-%COMP%]{-webkit-font-smoothing:auto;visibility:visible;opacity:1;z-index:1550}.pswp__top-bar[_ngcontent-%COMP%]{position:absolute;left:0;top:0;height:44px;width:100%}.pswp--has_mouse[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp--has_mouse[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%], .pswp__caption[_ngcontent-%COMP%], .pswp__top-bar[_ngcontent-%COMP%]{-webkit-backface-visibility:hidden;will-change:opacity;transition:opacity 333ms cubic-bezier(.4,0,.22,1)}.pswp--has_mouse[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp--has_mouse[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%]{visibility:visible}.pswp__caption[_ngcontent-%COMP%], .pswp__top-bar[_ngcontent-%COMP%]{background-color:rgba(0,0,0,.5)}.pswp__ui--fit[_ngcontent-%COMP%]   .pswp__caption[_ngcontent-%COMP%], .pswp__ui--fit[_ngcontent-%COMP%]   .pswp__top-bar[_ngcontent-%COMP%]{background-color:rgba(0,0,0,.3)}.pswp__ui--idle[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp__ui--idle[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%], .pswp__ui--idle[_ngcontent-%COMP%]   .pswp__top-bar[_ngcontent-%COMP%]{opacity:0}.pswp__ui--hidden[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp__ui--hidden[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%], .pswp__ui--hidden[_ngcontent-%COMP%]   .pswp__caption[_ngcontent-%COMP%], .pswp__ui--hidden[_ngcontent-%COMP%]   .pswp__top-bar[_ngcontent-%COMP%]{opacity:.001}.pswp__ui--one-slide[_ngcontent-%COMP%]   .pswp__button--arrow--left[_ngcontent-%COMP%], .pswp__ui--one-slide[_ngcontent-%COMP%]   .pswp__button--arrow--right[_ngcontent-%COMP%], .pswp__ui--one-slide[_ngcontent-%COMP%]   .pswp__counter[_ngcontent-%COMP%]{display:none}.pswp__element--disabled[_ngcontent-%COMP%]{display:none!important}.pswp--minimal--dark[_ngcontent-%COMP%]   .pswp__top-bar[_ngcontent-%COMP%]{background:0 0}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(LightboxComponent, [{
        type: i0.Component,
        args: [{
                selector: 'ngp-lightbox',
                template: "<!-- Root element of PhotoSwipe. Must have class pswp. -->\n<div class=\"pswp\" #ngpLightbox tabindex=\"-1\" role=\"dialog\" aria-hidden=\"true\">\n    <!-- Background of PhotoSwipe.\n         It's a separate element, as animating opacity is faster than rgba(). -->\n    <div class=\"pswp__bg\"></div>\n    <!-- Slides wrapper with overflow:hidden. -->\n    <div class=\"pswp__scroll-wrap\">\n        <!-- Container that holds slides. PhotoSwipe keeps only 3 slides in DOM to save memory. -->\n        <!-- don't modify these 3 pswp__item elements, data is added later on. -->\n        <div class=\"pswp__container\">\n            <div class=\"pswp__item\"></div>\n            <div class=\"pswp__item\"></div>\n            <div class=\"pswp__item\"></div>\n        </div>\n        <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->\n        <div class=\"pswp__ui pswp__ui--hidden\">\n            <div class=\"pswp__top-bar\">\n                <!--  Controls are self-explanatory. Order can be changed. -->\n                <div class=\"pswp__counter\"></div>\n                <button class=\"pswp__button pswp__button--close\" title=\"Close (Esc)\"></button>\n                <button class=\"pswp__button pswp__button--share\" title=\"Share\"></button>\n                <button class=\"pswp__button pswp__button--fs\" title=\"Toggle fullscreen\"></button>\n                <button class=\"pswp__button pswp__button--zoom\" title=\"Zoom in/out\"></button>\n                <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->\n                <!-- element will get class pswp__preloader--active when preloader is running -->\n                <div class=\"pswp__preloader\">\n                    <div class=\"pswp__preloader__icn\">\n                        <div class=\"pswp__preloader__cut\">\n                            <div class=\"pswp__preloader__donut\"></div>\n                        </div>\n                    </div>\n                </div>\n            </div>\n            <div class=\"pswp__share-modal pswp__share-modal--hidden pswp__single-tap\">\n                <div class=\"pswp__share-tooltip\"></div>\n            </div>\n            <button class=\"pswp__button pswp__button--arrow--left\" title=\"Previous (arrow left)\">\n            </button>\n            <button class=\"pswp__button pswp__button--arrow--right\" title=\"Next (arrow right)\">\n            </button>\n            <div class=\"pswp__caption\">\n                <div class=\"pswp__caption__center\"></div>\n            </div>\n        </div>\n    </div>\n</div>",
                styles: ["/*! PhotoSwipe main CSS by Dmitry Semenov | photoswipe.com | MIT license */.pswp{display:none;position:absolute;width:100%;height:100%;left:0;top:0;overflow:hidden;touch-action:none;z-index:1500;-webkit-text-size-adjust:100%;-webkit-backface-visibility:hidden;outline:0}.pswp *{box-sizing:border-box}.pswp img{max-width:none}.pswp--animate_opacity{opacity:.001;will-change:opacity;transition:opacity 333ms cubic-bezier(.4,0,.22,1)}.pswp--open{display:block}.pswp--zoom-allowed .pswp__img{cursor:-webkit-zoom-in;cursor:-moz-zoom-in;cursor:zoom-in}.pswp--zoomed-in .pswp__img{cursor:-webkit-grab;cursor:-moz-grab;cursor:grab}.pswp--dragging .pswp__img{cursor:-webkit-grabbing;cursor:-moz-grabbing;cursor:grabbing}.pswp__bg{position:absolute;left:0;top:0;width:100%;height:100%;background:#000;opacity:0;-webkit-transform:translateZ(0);transform:translateZ(0);-webkit-backface-visibility:hidden;will-change:opacity}.pswp__scroll-wrap{position:absolute;left:0;top:0;width:100%;height:100%;overflow:hidden}.pswp__container,.pswp__zoom-wrap{touch-action:none;position:absolute;left:0;right:0;top:0;bottom:0;-webkit-backface-visibility:hidden}.pswp__container,.pswp__img{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none}.pswp__zoom-wrap{position:absolute;width:100%;-webkit-transform-origin:left top;transform-origin:left top;transition:transform 333ms cubic-bezier(.4,0,.22,1),-webkit-transform 333ms cubic-bezier(.4,0,.22,1)}.pswp__bg{will-change:opacity;transition:opacity 333ms cubic-bezier(.4,0,.22,1)}.pswp--animated-in .pswp__bg,.pswp--animated-in .pswp__zoom-wrap{transition:none}.pswp__item{position:absolute;left:0;right:0;top:0;bottom:0;overflow:hidden}.pswp__img{position:absolute;width:auto;height:auto;top:0;left:0}.pswp__img--placeholder{-webkit-backface-visibility:hidden}.pswp__img--placeholder--blank{background:#222}.pswp--ie .pswp__img{width:100%!important;height:auto!important;left:0;top:0}.pswp__error-msg{position:absolute;left:0;top:50%;width:100%;text-align:center;font-size:14px;line-height:16px;margin-top:-8px;color:#ccc}.pswp__error-msg a{color:#ccc;text-decoration:underline}/*! PhotoSwipe Default UI CSS by Dmitry Semenov | photoswipe.com | MIT license */.pswp__button{position:relative;cursor:pointer;overflow:visible;-webkit-appearance:none;display:block;border:0;padding:0;margin:0;float:right;opacity:.75;transition:opacity .2s;box-shadow:none}.pswp__button:focus,.pswp__button:hover{opacity:1}.pswp__button:active{outline:0;opacity:.9}.pswp__button::-moz-focus-inner{padding:0;border:0}.pswp__ui--over-close .pswp__button--close{opacity:1}.pswp__button,.pswp__button--arrow--left:before,.pswp__button--arrow--right:before{background:url(/assets/media/default-skin.png) 0 0/264px 88px no-repeat;width:44px;height:44px}@media (-webkit-min-device-pixel-ratio:1.1),(-webkit-min-device-pixel-ratio:1.09375),(min-resolution:105dpi),(min-resolution:1.1dppx){.pswp--svg .pswp__button,.pswp--svg .pswp__button--arrow--left:before,.pswp--svg .pswp__button--arrow--right:before{background-image:url(/assets/media/default-skin.svg)}.pswp--svg .pswp__button--arrow--left,.pswp--svg .pswp__button--arrow--right{background:0 0}}.pswp__button--close{background-position:0 -44px}.pswp__button--share{background-position:-44px -44px}.pswp__button--fs{display:none}.pswp--supports-fs .pswp__button--fs{display:block}.pswp--fs .pswp__button--fs{background-position:-44px 0}.pswp__button--zoom{display:none;background-position:-88px 0}.pswp--zoom-allowed .pswp__button--zoom{display:block}.pswp--zoomed-in .pswp__button--zoom{background-position:-132px 0}.pswp--touch .pswp__button--arrow--left,.pswp--touch .pswp__button--arrow--right{visibility:hidden}.pswp__button--arrow--left,.pswp__button--arrow--right{background:0 0;top:50%;margin-top:-50px;width:70px;height:100px;position:absolute}.pswp__button--arrow--left{left:0}.pswp__button--arrow--right{right:0}.pswp__button--arrow--left:before,.pswp__button--arrow--right:before{content:'';top:35px;background-color:rgba(0,0,0,.3);height:30px;width:32px;position:absolute}.pswp__button--arrow--left:before{left:6px;background-position:-138px -44px}.pswp__button--arrow--right:before{right:6px;background-position:-94px -44px}.pswp__counter,.pswp__share-modal{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.pswp__share-modal{display:block;background:rgba(0,0,0,.5);width:100%;height:100%;top:0;left:0;padding:10px;position:absolute;z-index:1600;opacity:0;transition:opacity .25s ease-out;-webkit-backface-visibility:hidden;will-change:opacity}.pswp__share-modal--hidden{display:none}.pswp__share-tooltip{z-index:1620;position:absolute;background:#fff;top:56px;border-radius:2px;display:block;width:auto;right:44px;box-shadow:0 2px 5px rgba(0,0,0,.25);-webkit-transform:translateY(6px);transform:translateY(6px);transition:transform .25s;transition:transform .25s,-webkit-transform .25s;-webkit-backface-visibility:hidden;will-change:transform}.pswp__share-tooltip a{display:block;padding:8px 12px;color:#000;text-decoration:none;font-size:14px;line-height:18px}.pswp__share-tooltip a:hover{text-decoration:none;color:#000}.pswp__share-tooltip a:first-child{border-radius:2px 2px 0 0}.pswp__share-tooltip a:last-child{border-radius:0 0 2px 2px}.pswp__share-modal--fade-in{opacity:1}.pswp__share-modal--fade-in .pswp__share-tooltip{-webkit-transform:translateY(0);transform:translateY(0)}.pswp--touch .pswp__share-tooltip a{padding:16px 12px}a.pswp__share--facebook:before{content:'';display:block;width:0;height:0;position:absolute;top:-12px;right:15px;border:6px solid transparent;border-bottom-color:#fff;-webkit-pointer-events:none;-moz-pointer-events:none;pointer-events:none}a.pswp__share--facebook:hover{background:#3e5c9a;color:#fff}a.pswp__share--facebook:hover:before{border-bottom-color:#3e5c9a}a.pswp__share--twitter:hover{background:#55acee;color:#fff}a.pswp__share--pinterest:hover{background:#ccc;color:#ce272d}a.pswp__share--download:hover{background:#ddd}.pswp__counter{position:absolute;left:0;top:0;height:44px;font-size:13px;line-height:44px;color:#fff;opacity:.75;padding:0 10px}.pswp__caption{position:absolute;left:0;bottom:0;width:100%;min-height:44px}.pswp__caption small{font-size:11px;color:#bbb}.pswp__caption__center{text-align:left;max-width:420px;margin:0 auto;font-size:13px;padding:10px;line-height:20px;color:#ccc}.pswp__caption--empty{display:none}.pswp__caption--fake{visibility:hidden}.pswp__preloader{width:44px;height:44px;position:absolute;top:0;left:50%;margin-left:-22px;opacity:0;transition:opacity .25s ease-out;will-change:opacity;direction:ltr}.pswp__preloader__icn{width:20px;height:20px;margin:12px}.pswp__preloader--active{opacity:1}.pswp__preloader--active .pswp__preloader__icn{background:url(/assets/media/preloader.gif) no-repeat}.pswp--css_animation .pswp__preloader--active{opacity:1}.pswp--css_animation .pswp__preloader--active .pswp__preloader__icn{-webkit-animation:.5s linear infinite clockwise;animation:.5s linear infinite clockwise}.pswp--css_animation .pswp__preloader--active .pswp__preloader__donut{-webkit-animation:1s cubic-bezier(.4,0,.22,1) infinite donut-rotate;animation:1s cubic-bezier(.4,0,.22,1) infinite donut-rotate}.pswp--css_animation .pswp__preloader__icn{background:0 0;opacity:.75;width:14px;height:14px;position:absolute;left:15px;top:15px;margin:0}.pswp--css_animation .pswp__preloader__cut{position:relative;width:7px;height:14px;overflow:hidden}.pswp--css_animation .pswp__preloader__donut{box-sizing:border-box;width:14px;height:14px;border:2px solid #fff;border-radius:50%;border-left-color:transparent;border-bottom-color:transparent;position:absolute;top:0;left:0;background:0 0;margin:0}@media screen and (max-width:1024px){.pswp__preloader{position:relative;left:auto;top:auto;margin:0;float:right}}@-webkit-keyframes clockwise{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes clockwise{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes donut-rotate{0%,100%{-webkit-transform:rotate(0);transform:rotate(0)}50%{-webkit-transform:rotate(-140deg);transform:rotate(-140deg)}}@keyframes donut-rotate{0%,100%{-webkit-transform:rotate(0);transform:rotate(0)}50%{-webkit-transform:rotate(-140deg);transform:rotate(-140deg)}}.pswp__ui{-webkit-font-smoothing:auto;visibility:visible;opacity:1;z-index:1550}.pswp__top-bar{position:absolute;left:0;top:0;height:44px;width:100%}.pswp--has_mouse .pswp__button--arrow--left,.pswp--has_mouse .pswp__button--arrow--right,.pswp__caption,.pswp__top-bar{-webkit-backface-visibility:hidden;will-change:opacity;transition:opacity 333ms cubic-bezier(.4,0,.22,1)}.pswp--has_mouse .pswp__button--arrow--left,.pswp--has_mouse .pswp__button--arrow--right{visibility:visible}.pswp__caption,.pswp__top-bar{background-color:rgba(0,0,0,.5)}.pswp__ui--fit .pswp__caption,.pswp__ui--fit .pswp__top-bar{background-color:rgba(0,0,0,.3)}.pswp__ui--idle .pswp__button--arrow--left,.pswp__ui--idle .pswp__button--arrow--right,.pswp__ui--idle .pswp__top-bar{opacity:0}.pswp__ui--hidden .pswp__button--arrow--left,.pswp__ui--hidden .pswp__button--arrow--right,.pswp__ui--hidden .pswp__caption,.pswp__ui--hidden .pswp__top-bar{opacity:.001}.pswp__ui--one-slide .pswp__button--arrow--left,.pswp__ui--one-slide .pswp__button--arrow--right,.pswp__ui--one-slide .pswp__counter{display:none}.pswp__element--disabled{display:none!important}.pswp--minimal--dark .pswp__top-bar{background:0 0}"]
            }]
    }], function () { return [{ type: NgpService }]; }, { el: [{
            type: i0.ViewChild,
            args: ['ngpLightbox', { static: true }]
        }] }); })();
        return LightboxComponent;
    }());

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (b.hasOwnProperty(p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @return {?}
     */
    function ANGULAR2_PHOTO_SWIPE_FACTORY() {
        return new DefaultLightboxAdapter();
    }
    /**
     * @abstract
     */
    var LightboxAdapter = /** @class */ (function () {
        function LightboxAdapter() {
        }
        /** @nocollapse */ LightboxAdapter.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: ANGULAR2_PHOTO_SWIPE_FACTORY, token: LightboxAdapter, providedIn: "root" });
LightboxAdapter.ɵfac = function LightboxAdapter_Factory(t) { return new (t || LightboxAdapter)(); };
LightboxAdapter.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: LightboxAdapter, factory: function () { return ANGULAR2_PHOTO_SWIPE_FACTORY(); }, providedIn: 'root' });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(LightboxAdapter, [{
        type: i0.Injectable,
        args: [{ providedIn: 'root', useFactory: ANGULAR2_PHOTO_SWIPE_FACTORY }]
    }], function () { return []; }, null); })();
        return LightboxAdapter;
    }());
    var DefaultLightboxAdapter = /** @class */ (function (_super) {
        __extends(DefaultLightboxAdapter, _super);
        function DefaultLightboxAdapter() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this.allowPanToNext = true;
            _this.spacing = 0.12;
            _this.bgOpacity = 1.0;
            _this.mouseUsed = false;
            _this.loop = true;
            _this.pinchToClose = true;
            _this.closeOnScroll = true;
            _this.closeOnVerticalDrag = true;
            _this.hideAnimationDuration = 333;
            _this.showAnimationDuration = 333;
            _this.showHideOpacity = false;
            _this.escKey = true;
            _this.arrowKeys = true;
            return _this;
        }
DefaultLightboxAdapter.ɵfac = function DefaultLightboxAdapter_Factory(t) { return ɵDefaultLightboxAdapter_BaseFactory(t || DefaultLightboxAdapter); };
DefaultLightboxAdapter.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: DefaultLightboxAdapter, factory: function (t) { return DefaultLightboxAdapter.ɵfac(t); } });
var ɵDefaultLightboxAdapter_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(DefaultLightboxAdapter);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DefaultLightboxAdapter, [{
        type: i0.Injectable
    }], null, null); })();
        return DefaultLightboxAdapter;
    }(LightboxAdapter));

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var GalleryItemComponent = /** @class */ (function () {
        function GalleryItemComponent(adapter) {
            this.adapter = adapter;
            this.clicked = new i0.EventEmitter();
            this.isBootstrapEnabled = false;
            this.isBootstrapEnabled = this.adapter.enableBootstrap4;
        }
        /**
         * @return {?}
         */
        GalleryItemComponent.prototype.onClick = /**
         * @return {?}
         */
            function () {
                this.clicked.emit(this.image);
                return false;
            };
        /** @nocollapse */
        GalleryItemComponent.ctorParameters = function () {
            return [
                { type: LightboxAdapter }
            ];
        };
        GalleryItemComponent.propDecorators = {
            image: [{ type: i0.Input }],
            clicked: [{ type: i0.Output }],
            isBootstrapEnabled: [{ type: i0.HostBinding, args: ['class.card',] }]
        };
GalleryItemComponent.ɵfac = function GalleryItemComponent_Factory(t) { return new (t || GalleryItemComponent)(ɵngcc0.ɵɵdirectiveInject(LightboxAdapter)); };
GalleryItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GalleryItemComponent, selectors: [["ngp-gallery-item"]], hostVars: 2, hostBindings: function GalleryItemComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("card", ctx.isBootstrapEnabled);
    } }, inputs: { image: "image" }, outputs: { clicked: "clicked" }, decls: 2, vars: 8, consts: [["itemprop", "contentUrl", 3, "href", "click"], ["itemprop", "thumbnail", 3, "src", "ngClass", "alt"]], template: function GalleryItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "a", 0);
        ɵngcc0.ɵɵlistener("click", function GalleryItemComponent_Template_a_click_0_listener() { return ctx.onClick(); });
        ɵngcc0.ɵɵelement(1, "img", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵpropertyInterpolate("href", ctx.image.largeUrl, ɵngcc0.ɵɵsanitizeUrl);
        ɵngcc0.ɵɵattribute("data-size", ctx.image.width + "x" + ctx.image.height)("data-author", ctx.image.author);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵpropertyInterpolate("src", ctx.image.thumbUrl, ɵngcc0.ɵɵsanitizeUrl);
        ɵngcc0.ɵɵpropertyInterpolate("alt", ctx.image.description);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(6, _c1, ctx.isBootstrapEnabled));
    } }, directives: [ɵngcc1.NgClass], styles: [""] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(GalleryItemComponent, [{
        type: i0.Component,
        args: [{
                selector: 'ngp-gallery-item',
                template: "<a href=\"{{image.largeUrl}}\" itemprop=\"contentUrl\" [attr.data-size]=\"image.width + 'x' + image.height\"\n    [attr.data-author]=\"image.author\" (click)=\"onClick()\">\n    <img src=\"{{image.thumbUrl}}\" [ngClass]=\"{'card-img-top': isBootstrapEnabled }\"  itemprop=\"thumbnail\" alt=\"{{image.description}}\" />\n</a>",
                styles: [""]
            }]
    }], function () { return [{ type: LightboxAdapter }]; }, { clicked: [{
            type: i0.Output
        }], isBootstrapEnabled: [{
            type: i0.HostBinding,
            args: ['class.card']
        }], image: [{
            type: i0.Input
        }] }); })();
        return GalleryItemComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var GalleryComponent = /** @class */ (function () {
        function GalleryComponent(ngp, adapter) {
            this.ngp = ngp;
            this.adapter = adapter;
            this.shareLinkClick = new i0.EventEmitter();
            this.subscriptions = [];
            this.images = [];
            this.isBootstrapEnabled = this.adapter.enableBootstrap4;
        }
        /**
         * @return {?}
         */
        GalleryComponent.prototype.ngAfterContentInit = /**
         * @return {?}
         */
            function () {
                var _this_1 = this;
                this.images = ( /** @type {?} */(this.galleryItems.toArray().map(( /**
                 * @param {?} cp
                 * @return {?}
                 */function (cp) {
                    // listen for clicks;
                    _this_1.subscriptions.push(cp.clicked.subscribe(( /**
                     * @param {?} data
                     * @return {?}
                     */function (data) { return _this_1.onClick(data); })));
                    return cp.image;
                }))));
            };
        /**
         * @return {?}
         */
        GalleryComponent.prototype.ngOnDestroy = /**
         * @return {?}
         */
            function () {
                this.subscriptions.forEach(( /**
                 * @param {?} sub
                 * @return {?}
                 */function (sub) { return sub.unsubscribe(); }));
            };
        /**
         * @param {?} data
         * @return {?}
         */
        GalleryComponent.prototype.onClick = /**
         * @param {?} data
         * @return {?}
         */
            function (data) {
                this.openPhotoSwipe(data, this.galleryElement);
            };
        /**
         * @private
         * @param {?} img
         * @param {?} galleryDOM
         * @return {?}
         */
        GalleryComponent.prototype.openPhotoSwipe = /**
         * @private
         * @param {?} img
         * @param {?} galleryDOM
         * @return {?}
         */
            function (img, galleryDOM) {
                this.adapter.galleryUID = galleryDOM.nativeElement.getAttribute('data-pswp-uid');
                this.adapter.index = img.id;
                /** @type {?} */
                var PSWP = ( /** @type {?} */(this.ngp.LightboxElement.nativeElement));
                this.pswp = new PhotoSwipe(PSWP, PhotoSwipeUI_Default, this.getImagesAsPhotoswipe(), this.adapter);
                /** @type {?} */
                var _this = this;
                this.pswp.listen('shareLinkClick', ( /**
                 * @param {?} e
                 * @param {?} target
                 * @return {?}
                 */function (e, target) {
                    _this.shareLinkClick.emit({ e: e, target: target });
                }));
                this.pswp.init();
                return false;
            };
        /**
         * @private
         * @return {?}
         */
        GalleryComponent.prototype.getImagesAsPhotoswipe = /**
         * @private
         * @return {?}
         */
            function () {
                return this.images.map(( /**
                 * @param {?} image
                 * @return {?}
                 */function (image) {
                    return {
                        src: image.largeUrl,
                        w: image.width,
                        h: image.height,
                        pid: image.id,
                        title: image.description,
                        author: image.author
                    };
                }));
            };
        /** @nocollapse */
        GalleryComponent.ctorParameters = function () {
            return [
                { type: NgpService },
                { type: LightboxAdapter }
            ];
        };
        GalleryComponent.propDecorators = {
            galleryElement: [{ type: i0.ViewChild, args: ['ngpGallery', { static: true },] }],
            galleryItems: [{ type: i0.ContentChildren, args: [GalleryItemComponent,] }],
            galleryId: [{ type: i0.Input }],
            shareLinkClick: [{ type: i0.Output }]
        };
GalleryComponent.ɵfac = function GalleryComponent_Factory(t) { return new (t || GalleryComponent)(ɵngcc0.ɵɵdirectiveInject(NgpService), ɵngcc0.ɵɵdirectiveInject(LightboxAdapter)); };
GalleryComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GalleryComponent, selectors: [["ngp-gallery"]], contentQueries: function GalleryComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, GalleryItemComponent, 0);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.galleryItems = _t);
    } }, viewQuery: function GalleryComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c2, 3);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.galleryElement = _t.first);
    } }, inputs: { galleryId: "galleryId" }, outputs: { shareLinkClick: "shareLinkClick" }, ngContentSelectors: _c5, decls: 3, vars: 3, consts: [["id", "sampleId", "itemscope", "", "itemtype", "http://schema.org/ImageGallery", 1, "angular2_photoswipe", 3, "ngClass"], ["ngpGallery", ""]], template: function GalleryComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c3);
        ɵngcc0.ɵɵelementStart(0, "div", 0, 1);
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(1, _c4, ctx.isBootstrapEnabled));
    } }, directives: [ɵngcc1.NgClass], styles: [""] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(GalleryComponent, [{
        type: i0.Component,
        args: [{
                selector: 'ngp-gallery',
                template: "<div class=\"angular2_photoswipe\" [ngClass]=\"{'card-columns': isBootstrapEnabled }\"  #ngpGallery id=\"sampleId\" itemscope itemtype=\"http://schema.org/ImageGallery\">\n  <ng-content select=\"ngp-gallery-item\"></ng-content>\n</div>",
                styles: [""]
            }]
    }], function () { return [{ type: NgpService }, { type: LightboxAdapter }]; }, { shareLinkClick: [{
            type: i0.Output
        }], galleryElement: [{
            type: i0.ViewChild,
            args: ['ngpGallery', { static: true }]
        }], galleryItems: [{
            type: i0.ContentChildren,
            args: [GalleryItemComponent]
        }], galleryId: [{
            type: i0.Input
        }] }); })();
        return GalleryComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var Angular2PhotoswipeModule = /** @class */ (function () {
        function Angular2PhotoswipeModule() {
        }
Angular2PhotoswipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: Angular2PhotoswipeModule });
Angular2PhotoswipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function Angular2PhotoswipeModule_Factory(t) { return new (t || Angular2PhotoswipeModule)(); }, imports: [[common.CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(Angular2PhotoswipeModule, { declarations: [Angular2PhotoswipeComponent, LightboxComponent, GalleryComponent, GalleryItemComponent], imports: [ɵngcc1.CommonModule], exports: [LightboxComponent, GalleryComponent, GalleryItemComponent] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Angular2PhotoswipeModule, [{
        type: i0.NgModule,
        args: [{
                imports: [common.CommonModule],
                declarations: [Angular2PhotoswipeComponent, LightboxComponent, GalleryComponent, GalleryItemComponent],
                exports: [LightboxComponent, GalleryComponent, GalleryItemComponent]
            }]
    }], function () { return []; }, null); })();
        return Angular2PhotoswipeModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var Image = /** @class */ (function () {
        function Image() {
        }
        return Image;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.Angular2PhotoswipeService = Angular2PhotoswipeService;
    exports.Angular2PhotoswipeComponent = Angular2PhotoswipeComponent;
    exports.LightboxAdapter = LightboxAdapter;
    exports.Angular2PhotoswipeModule = Angular2PhotoswipeModule;
    exports.Image = Image;
    exports.ɵf = GalleryItemComponent;
    exports.ɵe = GalleryComponent;
    exports.ɵa = ANGULAR2_PHOTO_SWIPE_FACTORY;
    exports.ɵb = DefaultLightboxAdapter;
    exports.ɵc = LightboxComponent;
    exports.ɵd = NgpService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
