// angular2_photoswipe/esm2015/lib/angular2-photoswipe.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { Angular2PhotoswipeComponent } from './angular2-photoswipe.component';
import { LightboxComponent } from './lightbox/lightbox.component';
import { GalleryComponent } from './gallery/gallery.component';
import { GalleryItemComponent } from './gallery-item/gallery-item.component';
import * as ɵngcc0 from '@angular/core';
export { LightboxAdapter } from './lightbox-adapter';
export class Angular2PhotoswipeModule {
}
Angular2PhotoswipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: Angular2PhotoswipeModule });
Angular2PhotoswipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function Angular2PhotoswipeModule_Factory(t) { return new (t || Angular2PhotoswipeModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(Angular2PhotoswipeModule, { declarations: function () { return [Angular2PhotoswipeComponent, LightboxComponent, GalleryComponent, GalleryItemComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [LightboxComponent, GalleryComponent, GalleryItemComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Angular2PhotoswipeModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [Angular2PhotoswipeComponent, LightboxComponent, GalleryComponent, GalleryItemComponent],
                exports: [LightboxComponent, GalleryComponent, GalleryItemComponent]
            }]
    }], null, null); })();
