// angular2_photoswipe/esm2015/lib/angular2-photoswipe.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class Angular2PhotoswipeComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
Angular2PhotoswipeComponent.ɵfac = function Angular2PhotoswipeComponent_Factory(t) { return new (t || Angular2PhotoswipeComponent)(); };
Angular2PhotoswipeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: Angular2PhotoswipeComponent, selectors: [["ngp-angular2-photoswipe"]], decls: 2, vars: 0, template: function Angular2PhotoswipeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "p");
        ɵngcc0.ɵɵtext(1, " angular2-photoswipe works! ");
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
/** @nocollapse */
Angular2PhotoswipeComponent.ctorParameters = () => [];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Angular2PhotoswipeComponent, [{
        type: Component,
        args: [{
                selector: 'ngp-angular2-photoswipe',
                template: `
    <p>
      angular2-photoswipe works!
    </p>
  `
            }]
    }], function () { return []; }, null); })();
