// @clr/angular/esm2015/modal/modal-body.js
/*
 * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, HostListener } from '@angular/core';
/**
 * Allows modal overflow area to be scrollable via keyboard.
 * The modal body will focus with keyboard navigation only.
 * This allows inner focusable items to be focused without
 * the overflow scroll being focused.
 */
import * as ɵngcc0 from '@angular/core';
export class ClrModalBody {
    constructor() {
        this.tabindex = 0;
    }
    mouseDown() {
        // tabindex = 0 binding should be removed
        // so it won't be focused when click starts with mousedown
        delete this.tabindex;
    }
    mouseUp() {
        // set the tabindex binding back when click is completed with mouseup
        this.tabindex = 0;
    }
}
ClrModalBody.ɵfac = function ClrModalBody_Factory(t) { return new (t || ClrModalBody)(); };
ClrModalBody.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ClrModalBody, selectors: [["", 8, "modal-body"]], hostVars: 1, hostBindings: function ClrModalBody_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("mousedown", function ClrModalBody_mousedown_HostBindingHandler() { return ctx.mouseDown(); })("mouseup", function ClrModalBody_mouseup_HostBindingHandler() { return ctx.mouseUp(); });
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("tabindex", ctx.tabindex);
    } } });
ClrModalBody.propDecorators = {
    mouseDown: [{ type: HostListener, args: ['mousedown',] }],
    mouseUp: [{ type: HostListener, args: ['mouseup',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ClrModalBody, [{
        type: Directive,
        args: [{
                selector: '.modal-body',
                host: {
                    '[attr.tabindex]': 'tabindex'
                }
            }]
    }], function () { return []; }, { mouseDown: [{
            type: HostListener,
            args: ['mousedown']
        }], mouseUp: [{
            type: HostListener,
            args: ['mouseup']
        }] }); })();
