// ngx-datatable-footer/esm5/lib/datatable-footer/datatable-footer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter, ViewChild, TemplateRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../datatable-pager/datatable-pager.component';

function DatatableFooterComponent_ng_template_0_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 3);
} if (rf & 2) {
    var ctx_r8 = ɵngcc0.ɵɵnextContext();
    var offset_r5 = ctx_r8.offset;
    var pageSize_r2 = ctx_r8.pageSize;
    var rowCount_r1 = ctx_r8.rowCount;
    ɵngcc0.ɵɵproperty("innerHTML", "\u5F53\u524D " + ((offset_r5 || 0) * pageSize_r2 + 1) + " - " + ((offset_r5 + 1) * pageSize_r2 > rowCount_r1 ? rowCount_r1 : (offset_r5 + 1 || 1) * pageSize_r2) + " / " + rowCount_r1 + " \u884C", ɵngcc0.ɵɵsanitizeHtml);
} }
function DatatableFooterComponent_ng_template_0_app_datatable_pager_1_Template(rf, ctx) { if (rf & 1) {
    var _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "app-datatable-pager", 4);
    ɵngcc0.ɵɵlistener("change", function DatatableFooterComponent_ng_template_0_app_datatable_pager_1_Template_app_datatable_pager_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); var ctx_r9 = ɵngcc0.ɵɵnextContext(2); ctx_r9.datatable.onFooterPage($event); return ctx_r9.emitPage($event); })("pageChange", function DatatableFooterComponent_ng_template_0_app_datatable_pager_1_Template_app_datatable_pager_pageChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); var ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.emitPage($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r12 = ɵngcc0.ɵɵnextContext();
    var curPage_r4 = ctx_r12.curPage;
    var pageSize_r2 = ctx_r12.pageSize;
    var rowCount_r1 = ctx_r12.rowCount;
    var ctx_r7 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("linage", ctx_r7.linage)("displayBothEnds", ctx_r7.displayBothEnds)("pageNumber", ctx_r7.pageNumber)("pageSizes", ctx_r7.pageSizes)("pagerPreviousIcon", ctx_r7.pagerPreviousIcon)("pagerNextIcon", ctx_r7.pagerNextIcon)("pagerLeftArrowIcon", ctx_r7.pagerLeftArrowIcon)("pagerRightArrowIcon", ctx_r7.pagerRightArrowIcon)("page", curPage_r4)("size", pageSize_r2)("datatable", ctx_r7.datatable)("count", rowCount_r1);
} }
function DatatableFooterComponent_ng_template_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, DatatableFooterComponent_ng_template_0_div_0_Template, 1, 1, "div", 1);
    ɵngcc0.ɵɵtemplate(1, DatatableFooterComponent_ng_template_0_app_datatable_pager_1_Template, 1, 12, "app-datatable-pager", 2);
} if (rf & 2) {
    var rowCount_r1 = ctx.rowCount;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", rowCount_r1 > 0 && ctx_r0.totalNumber);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", rowCount_r1 > 0);
} }
var DatatableFooterComponent = /** @class */ (function () {
    function DatatableFooterComponent() {
        this.pagerLeftArrowIcon = 'fa fa-angle-left';
        this.pagerRightArrowIcon = 'fa fa-angle-right';
        this.pagerPreviousIcon = 'fa fa-step-backward';
        this.pagerNextIcon = 'fa fa-step-forward';
        this.pageSizes = [10, 20, 30, 40, 50];
        this.totalNumber = true;
        this.linage = true;
        this.displayBothEnds = true;
        this.pageNumber = true;
        this.pageChange = new EventEmitter();
    }
    Object.defineProperty(DatatableFooterComponent.prototype, "template", {
        get: /**
         * @return {?}
         */
        function () {
            return this._template;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._template = val;
            this.update();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DatatableFooterComponent.prototype, "datatable", {
        get: /**
         * @return {?}
         */
        function () {
            return this._datatable;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._datatable = val;
            this.update();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    DatatableFooterComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    DatatableFooterComponent.prototype.update = /**
     * @return {?}
     */
    function () {
        if (this._datatable &&
            this._template &&
            (!this._datatable.footer || !this._datatable.footer.template)) {
            if (this._datatable.footer) {
                this._datatable.footer.template = this._template;
            }
            else {
                this._datatable.footer = {
                    footerHeight: this._datatable.footerHeight,
                    totalMessage: this._datatable.messages.totalMessage,
                    selectedMessage: this._datatable.messages.selectedMessage,
                    pagerLeftArrowIcon: '',
                    pagerRightArrowIcon: '',
                    pagerPreviousIcon: '',
                    pagerNextIcon: '',
                    template: this._template
                };
            }
        }
    };
    /**
     * @param {?} page
     * @return {?}
     */
    DatatableFooterComponent.prototype.emitPage = /**
     * @param {?} page
     * @return {?}
     */
    function (page) {
        this.pageChange.emit(page);
    };
    /** @nocollapse */
    DatatableFooterComponent.ctorParameters = function () { return []; };
    DatatableFooterComponent.propDecorators = {
        template: [{ type: ViewChild, args: [TemplateRef,] }],
        datatable: [{ type: Input }],
        pagerLeftArrowIcon: [{ type: Input }],
        pagerRightArrowIcon: [{ type: Input }],
        pagerPreviousIcon: [{ type: Input }],
        pagerNextIcon: [{ type: Input }],
        pageSizes: [{ type: Input }],
        totalNumber: [{ type: Input }],
        linage: [{ type: Input }],
        displayBothEnds: [{ type: Input }],
        pageNumber: [{ type: Input }],
        page: [{ type: Input }],
        pageChange: [{ type: Output }]
    };
DatatableFooterComponent.ɵfac = function DatatableFooterComponent_Factory(t) { return new (t || DatatableFooterComponent)(); };
DatatableFooterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DatatableFooterComponent, selectors: [["ngx-datatable-footer"]], viewQuery: function DatatableFooterComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(TemplateRef, 1);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
    } }, inputs: { pagerLeftArrowIcon: "pagerLeftArrowIcon", pagerRightArrowIcon: "pagerRightArrowIcon", pagerPreviousIcon: "pagerPreviousIcon", pagerNextIcon: "pagerNextIcon", pageSizes: "pageSizes", totalNumber: "totalNumber", linage: "linage", displayBothEnds: "displayBothEnds", pageNumber: "pageNumber", datatable: "datatable", page: "page" }, outputs: { pageChange: "pageChange" }, decls: 1, vars: 0, consts: [["ngx-datatable-footer-template", ""], ["class", "page-count", 3, "innerHTML", 4, "ngIf"], ["class", "datatable-pager", 3, "linage", "displayBothEnds", "pageNumber", "pageSizes", "pagerPreviousIcon", "pagerNextIcon", "pagerLeftArrowIcon", "pagerRightArrowIcon", "page", "size", "datatable", "count", "change", "pageChange", 4, "ngIf"], [1, "page-count", 3, "innerHTML"], [1, "datatable-pager", 3, "linage", "displayBothEnds", "pageNumber", "pageSizes", "pagerPreviousIcon", "pagerNextIcon", "pagerLeftArrowIcon", "pagerRightArrowIcon", "page", "size", "datatable", "count", "change", "pageChange"]], template: function DatatableFooterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, DatatableFooterComponent_ng_template_0_Template, 2, 2, "ng-template", 0);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.DatatablePagerComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DatatableFooterComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-datatable-footer',
                template: "\n  <ng-template ngx-datatable-footer-template let-rowCount=\"rowCount\" let-pageSize=\"pageSize\" let-selectedCount=\"selectedCount\"\n    let-curPage=\"curPage\" let-offset=\"offset\">\n    <div class=\"page-count\" *ngIf=\"rowCount>0&&totalNumber\" [innerHTML]=\"'\u5F53\u524D '+((offset||0) * pageSize + 1)+' - '+((offset+1) * pageSize > rowCount?rowCount : ((offset+1)||1) * pageSize)+' / '+rowCount+' \u884C'\">\n    </div>\n    <app-datatable-pager *ngIf=\"rowCount>0\" [linage]=\"linage\" [displayBothEnds]=\"displayBothEnds\" [pageNumber]=\"pageNumber\" [pageSizes]=\"pageSizes\" [pagerPreviousIcon]=\"pagerPreviousIcon\" [pagerNextIcon]=\"pagerNextIcon\"\n      [pagerLeftArrowIcon]=\"pagerLeftArrowIcon\" [pagerRightArrowIcon]=\"pagerRightArrowIcon\" [page]=\"curPage\" [size]=\"pageSize\"\n      [datatable]=\"datatable\" [count]=\"rowCount\" (change)=\"datatable.onFooterPage($event);emitPage($event)\" (pageChange)=\"emitPage($event)\" class=\"datatable-pager\">\n    </app-datatable-pager>\n  </ng-template>"
            }]
    }], function () { return []; }, { pagerLeftArrowIcon: [{
            type: Input
        }], pagerRightArrowIcon: [{
            type: Input
        }], pagerPreviousIcon: [{
            type: Input
        }], pagerNextIcon: [{
            type: Input
        }], pageSizes: [{
            type: Input
        }], totalNumber: [{
            type: Input
        }], linage: [{
            type: Input
        }], displayBothEnds: [{
            type: Input
        }], pageNumber: [{
            type: Input
        }], pageChange: [{
            type: Output
        }], template: [{
            type: ViewChild,
            args: [TemplateRef]
        }], datatable: [{
            type: Input
        }], page: [{
            type: Input
        }] }); })();
    return DatatableFooterComponent;
}());
export { DatatableFooterComponent };
if (false) {
    /** @type {?} */
    DatatableFooterComponent.prototype.pagerLeftArrowIcon;
    /** @type {?} */
    DatatableFooterComponent.prototype.pagerRightArrowIcon;
    /** @type {?} */
    DatatableFooterComponent.prototype.pagerPreviousIcon;
    /** @type {?} */
    DatatableFooterComponent.prototype.pagerNextIcon;
    /** @type {?} */
    DatatableFooterComponent.prototype.pageSizes;
    /** @type {?} */
    DatatableFooterComponent.prototype.totalNumber;
    /** @type {?} */
    DatatableFooterComponent.prototype.linage;
    /** @type {?} */
    DatatableFooterComponent.prototype.displayBothEnds;
    /** @type {?} */
    DatatableFooterComponent.prototype.pageNumber;
    /** @type {?} */
    DatatableFooterComponent.prototype.page;
    /** @type {?} */
    DatatableFooterComponent.prototype.pageChange;
    /**
     * @type {?}
     * @private
     */
    DatatableFooterComponent.prototype._template;
    /**
     * @type {?}
     * @private
     */
    DatatableFooterComponent.prototype._datatable;
}
