// ngx-highlightjs/esm5/lib/highlight.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { Highlight } from './highlight';
import * as ɵngcc0 from '@angular/core';
var HighlightModule = /** @class */ (function () {
    function HighlightModule() {
    }
HighlightModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: HighlightModule });
HighlightModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function HighlightModule_Factory(t) { return new (t || HighlightModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(HighlightModule, { declarations: function () { return [Highlight]; }, exports: function () { return [Highlight]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(HighlightModule, [{
        type: NgModule,
        args: [{
                declarations: [Highlight],
                exports: [Highlight]
            }]
    }], function () { return []; }, null); })();
    return HighlightModule;
}());
export { HighlightModule };
