// angular2_photoswipe/esm5/lib/lightbox-adapter.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @return {?}
 */
import * as ɵngcc0 from '@angular/core';
export function ANGULAR2_PHOTO_SWIPE_FACTORY() {
    return new DefaultLightboxAdapter();
}
/**
 * @abstract
 */
var LightboxAdapter = /** @class */ (function () {
    function LightboxAdapter() {
    }
    /** @nocollapse */ LightboxAdapter.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: ANGULAR2_PHOTO_SWIPE_FACTORY, token: LightboxAdapter, providedIn: "root" });
LightboxAdapter.ɵfac = function LightboxAdapter_Factory(t) { return new (t || LightboxAdapter)(); };
LightboxAdapter.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: LightboxAdapter, factory: function () { return ANGULAR2_PHOTO_SWIPE_FACTORY(); }, providedIn: 'root' });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(LightboxAdapter, [{
        type: Injectable,
        args: [{ providedIn: 'root', useFactory: ANGULAR2_PHOTO_SWIPE_FACTORY }]
    }], function () { return []; }, null); })();
    return LightboxAdapter;
}());
export { LightboxAdapter };
if (false) {
    /** @type {?} */
    LightboxAdapter.prototype.getThumbBoundsFn;
    /** @type {?} */
    LightboxAdapter.prototype.getDoubleTapZoom;
    /** @type {?} */
    LightboxAdapter.prototype.isClickableElement;
    /** @type {?} */
    LightboxAdapter.prototype.addCaptionHTMLFn;
    /** @type {?} */
    LightboxAdapter.prototype.index;
    /** @type {?} */
    LightboxAdapter.prototype.showHideOpacity;
    /** @type {?} */
    LightboxAdapter.prototype.showAnimationDuration;
    /** @type {?} */
    LightboxAdapter.prototype.hideAnimationDuration;
    /** @type {?} */
    LightboxAdapter.prototype.bgOpacity;
    /** @type {?} */
    LightboxAdapter.prototype.spacing;
    /** @type {?} */
    LightboxAdapter.prototype.allowPanToNext;
    /** @type {?} */
    LightboxAdapter.prototype.maxSpreadZoom;
    /** @type {?} */
    LightboxAdapter.prototype.loop;
    /** @type {?} */
    LightboxAdapter.prototype.pinchToClose;
    /** @type {?} */
    LightboxAdapter.prototype.closeOnScroll;
    /** @type {?} */
    LightboxAdapter.prototype.closeOnVerticalDrag;
    /** @type {?} */
    LightboxAdapter.prototype.mouseUsed;
    /** @type {?} */
    LightboxAdapter.prototype.escKey;
    /** @type {?} */
    LightboxAdapter.prototype.arrowKeys;
    /** @type {?} */
    LightboxAdapter.prototype.history;
    /** @type {?} */
    LightboxAdapter.prototype.galleryUID;
    /** @type {?} */
    LightboxAdapter.prototype.galleryPIDs;
    /** @type {?} */
    LightboxAdapter.prototype.errorMsg;
    /** @type {?} */
    LightboxAdapter.prototype.barsSize;
    /** @type {?} */
    LightboxAdapter.prototype.timeToIdle;
    /** @type {?} */
    LightboxAdapter.prototype.timeToIdleOutside;
    /** @type {?} */
    LightboxAdapter.prototype.loadingIndicatorDelay;
    /** @type {?} */
    LightboxAdapter.prototype.closeEl;
    /** @type {?} */
    LightboxAdapter.prototype.captionEl;
    /** @type {?} */
    LightboxAdapter.prototype.fullscreenEl;
    /** @type {?} */
    LightboxAdapter.prototype.zoomEl;
    /** @type {?} */
    LightboxAdapter.prototype.shareEl;
    /** @type {?} */
    LightboxAdapter.prototype.counterEl;
    /** @type {?} */
    LightboxAdapter.prototype.arrowEl;
    /** @type {?} */
    LightboxAdapter.prototype.preloaderEl;
    /** @type {?} */
    LightboxAdapter.prototype.tapToClose;
    /** @type {?} */
    LightboxAdapter.prototype.tapToToggleControls;
    /** @type {?} */
    LightboxAdapter.prototype.clickToCloseNonZoomable;
    /** @type {?} */
    LightboxAdapter.prototype.closeElClasses;
    /** @type {?} */
    LightboxAdapter.prototype.indexIndicatorSep;
    /** @type {?} */
    LightboxAdapter.prototype.shareButtons;
    /** @type {?} */
    LightboxAdapter.prototype.enableBootstrap4;
}
var DefaultLightboxAdapter = /** @class */ (function (_super) {
    tslib_1.__extends(DefaultLightboxAdapter, _super);
    function DefaultLightboxAdapter() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.allowPanToNext = true;
        _this.spacing = 0.12;
        _this.bgOpacity = 1.0;
        _this.mouseUsed = false;
        _this.loop = true;
        _this.pinchToClose = true;
        _this.closeOnScroll = true;
        _this.closeOnVerticalDrag = true;
        _this.hideAnimationDuration = 333;
        _this.showAnimationDuration = 333;
        _this.showHideOpacity = false;
        _this.escKey = true;
        _this.arrowKeys = true;
        return _this;
    }
DefaultLightboxAdapter.ɵfac = function DefaultLightboxAdapter_Factory(t) { return ɵDefaultLightboxAdapter_BaseFactory(t || DefaultLightboxAdapter); };
DefaultLightboxAdapter.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: DefaultLightboxAdapter, factory: function (t) { return DefaultLightboxAdapter.ɵfac(t); } });
var ɵDefaultLightboxAdapter_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(DefaultLightboxAdapter);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DefaultLightboxAdapter, [{
        type: Injectable
    }], null, null); })();
    return DefaultLightboxAdapter;
}(LightboxAdapter));
export { DefaultLightboxAdapter };
if (false) {
    /** @type {?} */
    DefaultLightboxAdapter.prototype.allowPanToNext;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.spacing;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.bgOpacity;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.mouseUsed;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.loop;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.pinchToClose;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.closeOnScroll;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.closeOnVerticalDrag;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.hideAnimationDuration;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.showAnimationDuration;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.showHideOpacity;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.escKey;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.arrowKeys;
    /** @type {?} */
    DefaultLightboxAdapter.prototype.enableBootstrap4;
}
