// ngx-highlightjs/esm5/plus/code-from-url.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Pipe } from '@angular/core';
import { CodeLoader } from './code-loader';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './code-loader';
var CodeFromUrlPipe = /** @class */ (function () {
    function CodeFromUrlPipe(_loader) {
        this._loader = _loader;
    }
    /**
     * @param {?} url
     * @return {?}
     */
    CodeFromUrlPipe.prototype.transform = /**
     * @param {?} url
     * @return {?}
     */
    function (url) {
        return this._loader.getCodeFromUrl(url);
    };
    /** @nocollapse */
    CodeFromUrlPipe.ctorParameters = function () { return [
        { type: CodeLoader }
    ]; };
CodeFromUrlPipe.ɵfac = function CodeFromUrlPipe_Factory(t) { return new (t || CodeFromUrlPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CodeLoader)); };
CodeFromUrlPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "codeFromUrl", type: CodeFromUrlPipe, pure: true });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CodeFromUrlPipe, [{
        type: Pipe,
        args: [{
                name: 'codeFromUrl'
            }]
    }], function () { return [{ type: ɵngcc1.CodeLoader }]; }, null); })();
    return CodeFromUrlPipe;
}());
export { CodeFromUrlPipe };
if (false) {
    /**
     * @type {?}
     * @private
     */
    CodeFromUrlPipe.prototype._loader;
}
