// ngx-highlightjs/esm5/plus/gist.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, Pipe, Input, Output, EventEmitter } from '@angular/core';
import { CodeLoader } from './code-loader';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './code-loader';
var GistDirective = /** @class */ (function () {
    function GistDirective(_loader) {
        this._loader = _loader;
        this.gistLoad = new EventEmitter();
    }
    Object.defineProperty(GistDirective.prototype, "gist", {
        set: /**
         * @private
         * @param {?} value
         * @return {?}
         */
        function (value) {
            var _this = this;
            if (value) {
                this._loader.getCodeFromGist(value).subscribe((/**
                 * @param {?} gist
                 * @return {?}
                 */
                function (gist) { return _this.gistLoad.emit(gist); }));
            }
        },
        enumerable: true,
        configurable: true
    });
    /** @nocollapse */
    GistDirective.ctorParameters = function () { return [
        { type: CodeLoader }
    ]; };
    GistDirective.propDecorators = {
        gist: [{ type: Input }],
        gistLoad: [{ type: Output }]
    };
GistDirective.ɵfac = function GistDirective_Factory(t) { return new (t || GistDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CodeLoader)); };
GistDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: GistDirective, selectors: [["", "gist", ""]], inputs: { gist: "gist" }, outputs: { gistLoad: "gistLoad" } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(GistDirective, [{
        type: Directive,
        args: [{
                selector: '[gist]'
            }]
    }], function () { return [{ type: ɵngcc1.CodeLoader }]; }, { gistLoad: [{
            type: Output
        }], gist: [{
            type: Input
        }] }); })();
    return GistDirective;
}());
export { GistDirective };
if (false) {
    /** @type {?} */
    GistDirective.prototype.gistLoad;
    /**
     * @type {?}
     * @private
     */
    GistDirective.prototype._loader;
}
var GistFilePipe = /** @class */ (function () {
    function GistFilePipe() {
    }
    /**
     * @param {?} gist
     * @param {?} fileName
     * @return {?}
     */
    GistFilePipe.prototype.transform = /**
     * @param {?} gist
     * @param {?} fileName
     * @return {?}
     */
    function (gist, fileName) {
        return (gist && gist.files && gist.files[fileName]) ? gist.files[fileName].content : null;
    };
GistFilePipe.ɵfac = function GistFilePipe_Factory(t) { return new (t || GistFilePipe)(); };
GistFilePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "gistFile", type: GistFilePipe, pure: true });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(GistFilePipe, [{
        type: Pipe,
        args: [{
                name: 'gistFile'
            }]
    }], function () { return []; }, null); })();
    return GistFilePipe;
}());
export { GistFilePipe };
