// ngx-highlightjs/esm5/plus/highlight-plus.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';
import { HighlightModule } from 'ngx-highlightjs';
// Uncomment the following line for development
// import { HighlightModule } from '../../src/public-api';
import { GistFilePipe, GistDirective } from './gist';
import { CodeFromUrlPipe } from './code-from-url';
import * as ɵngcc0 from '@angular/core';
var HighlightPlusModule = /** @class */ (function () {
    function HighlightPlusModule() {
    }
HighlightPlusModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: HighlightPlusModule });
HighlightPlusModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function HighlightPlusModule_Factory(t) { return new (t || HighlightPlusModule)(); }, imports: [[
            HighlightModule,
            HttpClientModule
        ], HighlightModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(HighlightPlusModule, { declarations: function () { return [GistDirective,
        GistFilePipe,
        CodeFromUrlPipe]; }, imports: function () { return [HighlightModule,
        HttpClientModule]; }, exports: function () { return [HighlightModule,
        GistDirective,
        GistFilePipe,
        CodeFromUrlPipe]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(HighlightPlusModule, [{
        type: NgModule,
        args: [{
                imports: [
                    HighlightModule,
                    HttpClientModule
                ],
                declarations: [
                    GistDirective,
                    GistFilePipe,
                    CodeFromUrlPipe
                ],
                exports: [
                    HighlightModule,
                    GistDirective,
                    GistFilePipe,
                    CodeFromUrlPipe
                ]
            }]
    }], function () { return []; }, null); })();
    return HighlightPlusModule;
}());
export { HighlightPlusModule };
