// ngx-datatable-footer/esm5/lib/ngx-datatable-footer.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { DatatablePagerComponent } from './datatable-pager/datatable-pager.component';
import { DatatableFooterComponent } from './datatable-footer/datatable-footer.component';
import * as ɵngcc0 from '@angular/core';
var NgxDatatableFooterModule = /** @class */ (function () {
    function NgxDatatableFooterModule() {
    }
    /**
     * @return {?}
     */
    NgxDatatableFooterModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: NgxDatatableFooterModule
        };
    };
NgxDatatableFooterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxDatatableFooterModule });
NgxDatatableFooterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxDatatableFooterModule_Factory(t) { return new (t || NgxDatatableFooterModule)(); }, imports: [[
            CommonModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxDatatableFooterModule, { declarations: function () { return [DatatablePagerComponent,
        DatatableFooterComponent]; }, imports: function () { return [CommonModule,
        FormsModule]; }, exports: function () { return [DatatablePagerComponent,
        DatatableFooterComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxDatatableFooterModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule
                ],
                declarations: [
                    DatatablePagerComponent,
                    DatatableFooterComponent
                ],
                exports: [
                    DatatablePagerComponent,
                    DatatableFooterComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return NgxDatatableFooterModule;
}());
export { NgxDatatableFooterModule };
