// @clr/angular/esm2015/forms/common/success.js
/**
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Component, Optional } from '@angular/core';
import { ControlIdService } from './providers/control-id.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './providers/control-id.service';

const _c0 = ["*"];
export class ClrControlSuccess {
    constructor(controlIdService) {
        this.controlIdService = controlIdService;
    }
}
ClrControlSuccess.ɵfac = function ClrControlSuccess_Factory(t) { return new (t || ClrControlSuccess)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ControlIdService, 8)); };
ClrControlSuccess.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ClrControlSuccess, selectors: [["clr-control-success"]], hostVars: 3, hostBindings: function ClrControlSuccess_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵhostProperty("id", (ctx.controlIdService == null ? null : ctx.controlIdService.id) + "-success");
        ɵngcc0.ɵɵclassProp("clr-subtext", true);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function ClrControlSuccess_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
ClrControlSuccess.ctorParameters = () => [
    { type: ControlIdService, decorators: [{ type: Optional }] }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ClrControlSuccess, [{
        type: Component,
        args: [{
                selector: 'clr-control-success',
                template: ` <ng-content></ng-content> `,
                host: {
                    '[class.clr-subtext]': 'true',
                    '[id]': 'controlIdService?.id + "-success"'
                }
            }]
    }], function () { return [{ type: ɵngcc1.ControlIdService, decorators: [{
                type: Optional
            }] }]; }, null); })();
