// @clr/angular/esm2015/forms/input/input.module.js
/**
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { checkCircleIcon, ClarityIcons, exclamationCircleIcon } from '@cds/core/icon';
import { ClrIconModule } from '../../icon/icon.module';
import { ClrCommonFormsModule } from '../common/common.module';
import { ClrInput } from './input';
import { ClrInputContainer } from './input-container';
import * as ɵngcc0 from '@angular/core';
export class ClrInputModule {
    constructor() {
        ClarityIcons.addIcons(exclamationCircleIcon, checkCircleIcon // caret
        );
    }
}
ClrInputModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ClrInputModule });
ClrInputModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ClrInputModule_Factory(t) { return new (t || ClrInputModule)(); }, imports: [[CommonModule, FormsModule, ClrIconModule, ClrCommonFormsModule], ClrCommonFormsModule] });
ClrInputModule.ctorParameters = () => [];
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ClrInputModule, { declarations: function () { return [ClrInput, ClrInputContainer]; }, imports: function () { return [CommonModule, FormsModule, ClrIconModule, ClrCommonFormsModule]; }, exports: function () { return [ClrCommonFormsModule, ClrInput, ClrInputContainer]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ClrInputModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, ClrIconModule, ClrCommonFormsModule],
                declarations: [ClrInput, ClrInputContainer],
                exports: [ClrCommonFormsModule, ClrInput, ClrInputContainer],
                entryComponents: [ClrInputContainer]
            }]
    }], function () { return []; }, null); })();
