// ng-block-ui/esm2015/router/block-ui-prevent-navigation.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { BlockUIService, BLOCKUI_DEFAULT } from 'ng-block-ui';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'ng-block-ui';
let BlockUIPreventNavigation = class BlockUIPreventNavigation {
    constructor(blockUIService) {
        this.blockUIService = blockUIService;
    }
    canActivate() {
        return !this.blockUIService.isActive(BLOCKUI_DEFAULT);
    }
    canActivateChild() {
        return !this.blockUIService.isActive(BLOCKUI_DEFAULT);
    }
};
BlockUIPreventNavigation.ɵfac = function BlockUIPreventNavigation_Factory(t) { return new (t || BlockUIPreventNavigation)(ɵngcc0.ɵɵinject(ɵngcc1.BlockUIService)); };
BlockUIPreventNavigation.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: BlockUIPreventNavigation, factory: function (t) { return BlockUIPreventNavigation.ɵfac(t); } });
BlockUIPreventNavigation.ctorParameters = () => [
    { type: BlockUIService }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BlockUIPreventNavigation, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.BlockUIService }]; }, null); })();
export { BlockUIPreventNavigation };
