// ng-block-ui/esm2015/router/block-ui-router.module.js
import * as ɵngcc0 from '@angular/core';
var BlockUIRouterModule_1;
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { BlockUIModule } from 'ng-block-ui';
import { BlockUIPreventNavigation } from './block-ui-prevent-navigation.service';
let BlockUIRouterModule = BlockUIRouterModule_1 = class BlockUIRouterModule {
    static forRoot() {
        return {
            ngModule: BlockUIRouterModule_1,
            providers: [
                BlockUIPreventNavigation
            ]
        };
    }
};
BlockUIRouterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BlockUIRouterModule });
BlockUIRouterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BlockUIRouterModule_Factory(t) { return new (t || BlockUIRouterModule)(); }, imports: [[
            BlockUIModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BlockUIRouterModule, { imports: function () { return [BlockUIModule]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BlockUIRouterModule, [{
        type: NgModule,
        args: [{
                imports: [
                    BlockUIModule
                ]
            }]
    }], null, null); })();
export { BlockUIRouterModule };
