// ng-block-ui/esm5/router/block-ui-router.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { BlockUIModule } from 'ng-block-ui';
import { BlockUIPreventNavigation } from './block-ui-prevent-navigation.service';
import * as ɵngcc0 from '@angular/core';
var BlockUIRouterModule = /** @class */ (function () {
    function BlockUIRouterModule() {
    }
    BlockUIRouterModule_1 = BlockUIRouterModule;
    BlockUIRouterModule.forRoot = function () {
        return {
            ngModule: BlockUIRouterModule_1,
            providers: [
                BlockUIPreventNavigation
            ]
        };
    };
    var BlockUIRouterModule_1;
BlockUIRouterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BlockUIRouterModule });
BlockUIRouterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BlockUIRouterModule_Factory(t) { return new (t || BlockUIRouterModule)(); }, imports: [[
            BlockUIModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BlockUIRouterModule, { imports: function () { return [BlockUIModule]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BlockUIRouterModule, [{
        type: NgModule,
        args: [{
                imports: [
                    BlockUIModule
                ]
            }]
    }], function () { return []; }, null); })();
    return BlockUIRouterModule;
}());
export { BlockUIRouterModule };
