// angularx-flatpickr/esm2015/flatpickr.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule, InjectionToken } from '@angular/core';
import { FlatpickrDirective } from './flatpickr.directive';
import { FlatpickrDefaults } from './flatpickr-defaults.service';
import * as ɵngcc0 from '@angular/core';
export const /** @type {?} */ USER_DEFAULTS = new InjectionToken('flatpickr defaults');
/**
 * @param {?} userDefaults
 * @return {?}
 */
export function defaultsFactory(userDefaults) {
    const /** @type {?} */ defaults = new FlatpickrDefaults();
    Object.assign(defaults, userDefaults);
    return defaults;
}
export class FlatpickrModule {
    /**
     * @param {?=} userDefaults
     * @return {?}
     */
    static forRoot(userDefaults = {}) {
        return {
            ngModule: FlatpickrModule,
            providers: [
                {
                    provide: USER_DEFAULTS,
                    useValue: userDefaults
                },
                {
                    provide: FlatpickrDefaults,
                    useFactory: defaultsFactory,
                    deps: [USER_DEFAULTS]
                }
            ]
        };
    }
}
FlatpickrModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FlatpickrModule });
FlatpickrModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FlatpickrModule_Factory(t) { return new (t || FlatpickrModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FlatpickrModule, { declarations: function () { return [FlatpickrDirective]; }, exports: function () { return [FlatpickrDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FlatpickrModule, [{
        type: NgModule,
        args: [{
                declarations: [FlatpickrDirective],
                exports: [FlatpickrDirective]
            }]
    }], null, null); })();
