// ng-block-ui/esm5/components/block-ui/block-ui.component.js
import { __decorate } from "tslib";
import { Component, Input, ViewEncapsulation } from '@angular/core';
import { BlockUIInstanceService } from '../../services/block-ui-instance.service';
import { BlockUIDefaultName } from '../../constants/block-ui-default-name.constant';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/block-ui-instance.service';
import * as ɵngcc2 from '../block-ui-content/block-ui-content.component';

var _c0 = ["*"];
var BlockUIComponent = /** @class */ (function () {
    function BlockUIComponent(blockUI) {
        this.blockUI = blockUI;
    }
    BlockUIComponent.prototype.ngOnInit = function () {
        this.name = this.name || BlockUIDefaultName;
        this.template = this.template || this.blockUI.blockUISettings.template;
    };
    BlockUIComponent.ctorParameters = function () { return [
        { type: BlockUIInstanceService }
    ]; };
    __decorate([
        Input()
    ], BlockUIComponent.prototype, "name", void 0);
    __decorate([
        Input()
    ], BlockUIComponent.prototype, "message", void 0);
    __decorate([
        Input()
    ], BlockUIComponent.prototype, "delayStart", void 0);
    __decorate([
        Input()
    ], BlockUIComponent.prototype, "delayStop", void 0);
    __decorate([
        Input()
    ], BlockUIComponent.prototype, "template", void 0);
BlockUIComponent.ɵfac = function BlockUIComponent_Factory(t) { return new (t || BlockUIComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.BlockUIInstanceService)); };
BlockUIComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BlockUIComponent, selectors: [["block-ui"]], inputs: { name: "name", template: "template", message: "message", delayStart: "delayStart", delayStop: "delayStop" }, ngContentSelectors: _c0, decls: 2, vars: 5, consts: [[3, "name", "message", "template", "delayStart", "delayStop"]], template: function BlockUIComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
        ɵngcc0.ɵɵelement(1, "block-ui-content", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("name", ctx.name)("message", ctx.message)("template", ctx.template)("delayStart", ctx.delayStart)("delayStop", ctx.delayStop);
    } }, directives: [ɵngcc2.BlockUIContentComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BlockUIComponent, [{
        type: Component,
        args: [{
                selector: 'block-ui',
                template: "\n    <ng-content></ng-content>\n    <block-ui-content\n      [name]=\"name\"\n      [message]=\"message\"\n      [template]=\"template\"\n      [delayStart]=\"delayStart\"\n      [delayStop]=\"delayStop\"\n    >\n    </block-ui-content>\n  ",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.BlockUIInstanceService }]; }, { name: [{
            type: Input
        }], template: [{
            type: Input
        }], message: [{
            type: Input
        }], delayStart: [{
            type: Input
        }], delayStop: [{
            type: Input
        }] }); })();
    return BlockUIComponent;
}());
export { BlockUIComponent };
