// angular2_photoswipe/esm2015/lib/gallery-item/gallery-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter, HostBinding } from '@angular/core';
import { Image } from '../image';
import { LightboxAdapter } from '../lightbox-adapter';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../lightbox-adapter';
import * as ɵngcc2 from '@angular/common';

const _c0 = function (a0) { return { "card-img-top": a0 }; };
export class GalleryItemComponent {
    /**
     * @param {?} adapter
     */
    constructor(adapter) {
        this.adapter = adapter;
        this.clicked = new EventEmitter();
        this.isBootstrapEnabled = false;
        this.isBootstrapEnabled = this.adapter.enableBootstrap4;
    }
    /**
     * @return {?}
     */
    onClick() {
        this.clicked.emit(this.image);
        return false;
    }
}
GalleryItemComponent.ɵfac = function GalleryItemComponent_Factory(t) { return new (t || GalleryItemComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.LightboxAdapter)); };
GalleryItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GalleryItemComponent, selectors: [["ngp-gallery-item"]], hostVars: 2, hostBindings: function GalleryItemComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("card", ctx.isBootstrapEnabled);
    } }, inputs: { image: "image" }, outputs: { clicked: "clicked" }, decls: 2, vars: 8, consts: [["itemprop", "contentUrl", 3, "href", "click"], ["itemprop", "thumbnail", 3, "src", "ngClass", "alt"]], template: function GalleryItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "a", 0);
        ɵngcc0.ɵɵlistener("click", function GalleryItemComponent_Template_a_click_0_listener() { return ctx.onClick(); });
        ɵngcc0.ɵɵelement(1, "img", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵpropertyInterpolate("href", ctx.image.largeUrl, ɵngcc0.ɵɵsanitizeUrl);
        ɵngcc0.ɵɵattribute("data-size", ctx.image.width + "x" + ctx.image.height)("data-author", ctx.image.author);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵpropertyInterpolate("src", ctx.image.thumbUrl, ɵngcc0.ɵɵsanitizeUrl);
        ɵngcc0.ɵɵpropertyInterpolate("alt", ctx.image.description);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(6, _c0, ctx.isBootstrapEnabled));
    } }, directives: [ɵngcc2.NgClass], styles: [""] });
/** @nocollapse */
GalleryItemComponent.ctorParameters = () => [
    { type: LightboxAdapter }
];
GalleryItemComponent.propDecorators = {
    image: [{ type: Input }],
    clicked: [{ type: Output }],
    isBootstrapEnabled: [{ type: HostBinding, args: ['class.card',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(GalleryItemComponent, [{
        type: Component,
        args: [{
                selector: 'ngp-gallery-item',
                template: "<a href=\"{{image.largeUrl}}\" itemprop=\"contentUrl\" [attr.data-size]=\"image.width + 'x' + image.height\"\n    [attr.data-author]=\"image.author\" (click)=\"onClick()\">\n    <img src=\"{{image.thumbUrl}}\" [ngClass]=\"{'card-img-top': isBootstrapEnabled }\"  itemprop=\"thumbnail\" alt=\"{{image.description}}\" />\n</a>",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc1.LightboxAdapter }]; }, { clicked: [{
            type: Output
        }], isBootstrapEnabled: [{
            type: HostBinding,
            args: ['class.card']
        }], image: [{
            type: Input
        }] }); })();
if (false) {
    /** @type {?} */
    GalleryItemComponent.prototype.image;
    /** @type {?} */
    GalleryItemComponent.prototype.clicked;
    /** @type {?} */
    GalleryItemComponent.prototype.isBootstrapEnabled;
    /**
     * @type {?}
     * @private
     */
    GalleryItemComponent.prototype.adapter;
}
